thorlcr/thorutil/thbuf.cpp

```diff
@@ -284,6 +284,8 @@ class CSmartRowBuffer: public CSimpleInterface, implements ISmartRowBuffer, impl
         while (out->ordinality()) 
             ReleaseThorRow(out->dequeue());
         delete out;
+        if (file)
```

Comment:
It did work, but you're quite right.
Should test/clear fileio 1st.
Will change.